froide/foirequest/api/serializers.py

```diff
@@ -157,13 +161,16 @@ def create(self, validated_data):
         return service.execute(validated_data["request"])
 
 
+class FoiRequestRelatedField(serializers.HyperlinkedRelatedField):
+    view_name = "api:request-detail"
+
+    def get_queryset(self):
+        return get_read_foirequest_queryset(self.context["request"])
+
+
 class FoiMessageSerializer(serializers.HyperlinkedModelSerializer):
-    resource_uri = serializers.HyperlinkedIdentityField(
-        view_name="api:message-detail", lookup_field="pk"
-    )
-    request = serializers.HyperlinkedRelatedField(
-        read_only=True, view_name="api:request-detail"
-    )
+    resource_uri = serializers.HyperlinkedIdentityField(view_name="api:message-detail")
+    request = FoiRequestRelatedField()
```

Comment:
is this intentional or should this be 
```suggestion
    request = FoiRequestRelatedField(read_only=True)
```

Reply:
Yes, since `read_only` also affects the `create` action. I've added a safeguard to prevent the field from being modified: [`eab4414` (#888)](https://github.com/okfde/froide/pull/888/commits/eab44143d4fba54048cee05f758d748e1bf0852c#diff-942b7081675f4bd4e08c833a5d7cc2cf9d6631a444a5d0419f5616c7a5dbf7e4R178-R181) (see also https://github.com/encode/django-rest-framework/discussions/8606)

Reply:
I think it might be better to work with an allowlist here (aka only allow changing this on creation). Does `update` also cover `partial_update`)?